### PR TITLE
Add benchmarking to the docs

### LINKED ISSUE #205: benchmarking — create a notebook showing how to work with the single gpu benchmarks
The notebook should

- show how to understand what the benchmark's options are
- show how to programmatically run the benchmark
- show how to create the callable that the benchmark is invoking
- show how to acquire the input generator the benchmark is using to create the inputs to the callable

This way practitioners will be able to run the benchmark callable in their own contexts, which should support profiling.

fyi @kevinstephano — is this what you were thinking of when talking about extracting a callable from a benchmark?

cc @borda @apaz-cli @crcrpar

### PR DESCRIPTION
**Pull request by riccardofelluga**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Closes #205 by adding the benchmarking to the docs.

### CODE REVIEW

File: docs/source/intermediate/benchmarking.rst [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+As an example, if you want to use cudnn as executor your terminal will looks something like::
+
+  python thunder/benchmarks/benchmark_litgpt.py --model_name Llama-2-7b-hf  --compile thunder_cudnn
+
+
+and if you are testing ``torch.compile`` then it will look something like this::
+
+  python thunder/benchmarks/benchmark_litgpt.py --model_name Llama-2-7b-hf  --compile inductor
+
+PyTest benchmarks
```
Comment (crcrpar) [inline docs/source/intermediate/benchmarking.rst:79]: Is this the official spelling of pytest?

File: docs/source/intermediate/benchmarking.rst [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+Now this is just half of the test, what about the backward pass? In this case, you'll need to implement a ``postprocess_for_backward()`` method to take care of that::
+
+  def postprocess_for_backward(self, out: torch.Tensor) -> torch.Tensor | None:
+    # Check if backward it's needed at all
+    if not self.requires_grad:
+      return
+
+    targets = make_tensor_like(out)  # fake targets
+    loss = torch.nn.functional.mse_loss(logits, targets)
+    return loss
```
Comment (crcrpar) [inline docs/source/intermediate/benchmarking.rst:224]: ditto

**Review by crcrpar — COMMENTED** [review]
this looks nice overall to me. I left some nit-picking comments

File: docs/source/intermediate/benchmarking.rst (hunk 1 above)
Comment (riccardofelluga) [reply docs/source/intermediate/benchmarking.rst:79]: No it's not, I took some liberty but you're right it's better to keep to the official spelling

**Review by t-vi — APPROVED** [review]
Thank you @riccardofelluga ! Love the style. Thank you @crcrpar for your review and comments!

**Pull request merged**